### PR TITLE
Add request module

### PR DESCRIPTION
**Pull request by abehnia**

### CODE REVIEW

**Comment by abehnia** [comment]
Request/response module that facilitates sending messages between BBs. Get requests and send requests depend only on types, they're resolved at compile them. A number of these can be put into an enum (the pattern matching will be the only thing resolved dynamically) and sent over the network.

**Review by brianmakesthings — APPROVED** [review]
Looks good to me but probably needs a second review

**Pull request merged**
